lib/Reader/readerir.cpp

```diff
@@ -5100,6 +5110,49 @@ IRNode *GenIR::convertToBoxHelperArgumentType(IRNode *Opr,
   return Opr;
 }
 
+// Method is called with empty stack.
+void GenIR::jmp(ReaderBaseNS::CallOpcode Opcode, mdToken Token) {
+  assert(Opcode == ReaderBaseNS::Jmp);
+  ReaderCallTargetData Data;
+  makeReaderCallTargetDataForJmp(&Data, Token);
+
+  const bool IsJmp = true;
+  if (MethodSignature.hasThis()) {
+    IRNode *ThisArg = loadArg(MethodSignature.getThisIndex(), IsJmp);
+    ReaderOperandStack->push(ThisArg);
+  }
+
+  for (uint32_t I = MethodSignature.getNormalParamStart();
+       I < MethodSignature.getNormalParamEnd(); ++I) {
+    IRNode *NormalArg = nullptr;
+    const ABIArgInfo ArgInfo = ABIMethodSig.getArgumentInfo(I);
+    if (ArgInfo.getKind() == ABIArgInfo::Indirect) {
+      // We pass indirect arguments without copying them to the current frame.
+      NormalArg = (IRNode *)Arguments[I];
+    } else {
+      NormalArg = loadArg(I, IsJmp);
+    }
+    ReaderOperandStack->push(NormalArg);
+  }
+
+  // This will generate a call marked with musttail.
+  IRNode *CallNode = nullptr;
+  rdrCall(&Data, Opcode, &CallNode);
```

Comment:
Will this push the result of the call on the stack? If so, I think we need to pop it (or update `genCall` to avoid generating a `ReturnNode` at all).